### DISCUSSION TITLE
Unable to retrieve JMAP identity

### DISCUSSION THREAD

**Discussion by igawagc** [stalwartlabs/jmap-server#66] [Q&A]
I apologize for this basic question.

I installed Stalwart Mail, registered the domain, user accounts, and email addresses.

Using the user account and password, I was able to create an email with Email/set via curl. However, when I tried to use Identity/get, I received the following error:

{"methodResponses":[["Identity/get",{"accountId":"p333333333333","state":"n","list":[],"notFound":[]},"0"]],"sessionState":"3e25b2a0"}

It seems that the identity could not be found.

I thought I might need to set the identity first, so I tried to use Identity/set with curl, but I encountered this error:

{"methodResponses":[["Identity/set",{"notCreated":{"k1":{"type":"invalidProperties","description":"E-mail address not configured for this account.","properties":["email"]}}},"0"]],"sessionState":"3e25b2a0"} 

The error indicates that the email address is not configured for this account.

The account is created, the email address is registered, and I can create emails. However, I keep getting an error saying that the email address is not registered when using Identity/set. What could be the cause of this issue?

**Comment by mdecimus** [comment]
The email address has to be listed in the user's account. This is to prevent users from adding addresses they are not authorized to send emails from.
From the webadmin go to Accounts and check which are the email addresses listed in this account.

**Comment by igawagc** [comment]
Thank you for your response.

In the Stalwart web administration panel, I have registered the account and its associated email address as shown in the attached file. I am trying to retrieve the Identity using this account and email address.

Are there any additional settings required besides adding the account and email address?
<img width="946" alt="2024-09-30 105123" src="https://github.com/user-attachments/assets/fd15b3ab-f26a-4efd-966e-73474aa85e77">

**Reply by mdecimus** [reply]
Having a valid email address is enough. Please note that the default identity is only created if there are no identities in the account.

**Comment by igawagc** [comment]
Thank you for your response.

Regarding the attached image, could you confirm that the account "igawa" has the email address "igawa@mail.atpopjmap.com" set as a valid email address?

However, when I try to retrieve the identity using the following command:

curl -u 'igawa:yama0306' \
     -X POST \
     --header "Content-Type: application/json" \
     --data '{
       "using": ["urn:ietf:params:jmap:core", "urn:ietf:params:jmap:mail"],
       "methodCalls": [
         [
           "Identity/get", 
           {}, 
           "0"
         ]
       ]
     }' \
     http://mail.atpopjmap.com:8080/jmap

I get the following error:

{"methodResponses":[["Identity/get",{"accountId":"p333333333333","state":"n","list":[],"notFound":[]},"0"]],"sessionState":"3e25b2a0"}

The identity cannot be found, as shown in the error.

I apologize for the repeated confirmation, but I would appreciate your further assistance.

**Comment by mdecimus** [comment]
> Regarding the attached image, could you confirm that the account "igawa" has the email address "[igawa@mail.atpopjmap.com](mailto:igawa@mail.atpopjmap.com)" set as a valid email address?

Yes.

> I get the following error:

You forgot to specify the `accountId`:

```sh
$ curl -k -u 'john@example.org:12345' \
-X POST \
--header "Content-Type: application/json" \
--data '{
"using": ["urn:ietf:params:jmap:core", "urn:ietf:params:jmap:mail"],
"methodCalls": [
[
"Identity/get",
{
"accountId": "bg",
},
"0"
]
]
}' \
https://localhost/jmap
```

```json
{"methodResponses":[["Identity/get",{"accountId":"bg","state":"n","list":[{"id":"a","name":"John Doe3","email":"john@example.org","replyTo":null,"bcc":null,"textSignature":"","htmlSignature":"","mayDelete":true}],"notFound":[]},"0"]],"sessionState":"3e25b2a0"}  
```
